src/sparseml/modifiers/obcq/utils/layer_compressor.py

```diff
@@ -0,0 +1,259 @@
+        else:
+            self.sequentially_compress(**kwargs)
+
+        extras = self.post_compress(**kwargs)
+
+        return {"outputs": extras["outputs"]}
+
+    def post_compress(self, **kwargs) -> Dict:
+        """
+        Clean up after compression
+
+        :return: outputs of the layer
+        """
+        nsamples = len(self.inputs)
+        outputs = []
+        forward_args_spec = inspect.getfullargspec(self.layer.__class__.forward)
+        passed_in_args = [arg for arg in forward_args_spec.args if arg in kwargs]
+        for j in range(nsamples):
+            passed_in_kwargs = {}
+            for arg in passed_in_args:
+                if isinstance(kwargs[arg], List):
+                    passed_in_kwargs[arg] = kwargs[arg][j]
+                else:
+                    passed_in_kwargs[arg] = kwargs[arg]
+            outputs.append(self.layer(self.inputs[j], **passed_in_kwargs)[0])
+
+        self.inputs = None
+        torch.cuda.empty_cache()
+        return {"outputs": outputs}
+
+    def sequentially_compress(self, **kwargs):
+        """
+        Run compression module by module, in dependency order
+        """
+        subset = self.compressible_modules()
+
+        forward_args_spec = inspect.getfullargspec(self.layer.__class__.forward)
+        passed_in_args = [arg for arg in forward_args_spec.args if arg in kwargs]
+
+        passed_in_kwargs = {}
+        for arg in passed_in_args:
+            if isinstance(kwargs[arg], List):
+                passed_in_kwargs[arg] = kwargs[arg][0]
+            else:
+                passed_in_kwargs[arg] = kwargs[arg]
+        order = _find_dependency_order(
+            self.layer, subset, self.inputs[0], **passed_in_kwargs
+        )
+
+        nsamples = len(self.inputs)
+        for name in order:
+            gpts = SparseGPT(subset[name])
+
+            def add_batch(name):
+                def tmp(_, inp, out):
+                    gpts.add_batch(inp[0].data, out.data)
+
+                return tmp
+
+            handle = subset[name].register_forward_hook(add_batch(name))
+            for j in range(nsamples):
+                passed_in_kwargs = {}
+                for arg in passed_in_args:
+                    if isinstance(kwargs[arg], List):
+                        passed_in_kwargs[arg] = kwargs[arg][j]
+                    else:
+                        passed_in_kwargs[arg] = kwargs[arg]
+                self.layer(self.inputs[j], **passed_in_kwargs)
+            handle.remove()
+
+            _LOGGER.info(f"Compressing module {name} of layer {self.layer_index}")
+            gpts.fasterprune(
+                self.args["sparsity"],
+                prunen=self.args["prunen"],
+                prunem=self.args["prunem"],
+                percdamp=self.args["percdamp"],
+                blocksize=self.args["blocksize"],
+            )
+            gpts.free()
+
+
+def _find_dependency_order(layer, subset, an_input, **kwargs):
+    order = []
```

Comment:
this is a cool function - might want to formalize a bit and move to our pytorch helpers